src/messages/mod.rs

```diff
@@ -122,6 +129,43 @@ impl Message {
         Ok(msg)
     }
 
+    /// Creates a message signed using a BLS KeyShare for destination accumulation
+    pub(crate) fn for_dst_accumulation(
+        node: &Node,
+        key_share: &SectionKeyShare,
+        dst: DstLocation,
+        variant: Variant,
```

Comment:
If I understand it correctly, we are currently using this dst accumulation only for `UserMessage`, are we? Although in theory we could use it for other variants as well, that's not how it's implemented in this PR, correct? So that's probably fine, but could be a potential source of confusion as it's now easy for someone to accidentally create invalid message. So why not make this restriction explicit? I'd suggest to change this function so that instead of taking `Variant`, it would take only `Bytes` and construct the `Variant::UserMessage` internally. Similarly, it seems the dst must be only `DstLocation::AccumulatingNode`. So again, construct it internally and only take the `XorName` here. That way the message should be correct by construction. Does it make sense?